velox/exec/HashTable.cpp

```diff
@@ -669,9 +671,8 @@ void HashTable<ignoreNullKeys>::checkSize(int32_t numNew) {
     // then the table lookup will become slow. Given that, we treat tombstone
     // slot as non-empty slot here to decide whether to trigger rehash or not.
   } else if (newNumDistincts > rehashSize()) {
-    // NOTE: we need to plus one here as newNumDistincts itself could be power
-    // of two.
-    auto newSize = bits::nextPowerOfTwo(newNumDistincts + 1);
+    auto newSize = bits::nextPowerOfTwo(stats_.capacity + 1);
+
```

Comment:
Can we have a unit test that
1. initial table size is N
2. keep adding new entries with one at a time till size N - 1
3. we expect the number of rehashes is 1?

The problem is that we do multiple rehash once the number of distincts exceeds rehashSize()? Thanks!

Reply:
I'd like to add this detailed unit test in a followup to remove the aggregation inefficiency sooner.
The unit test modified in this PR guarantees that the issue has been fixed.